### PR TITLE
chore: rotate NPM token

### PR DESCRIPTION
**Pull request by serhalp**
We recently revoked the NPM token used by this repo in CI. This rolls it to a new credential and update the naming and auth style at the same time.

For some reason this started failing on node 8. Node 8 is no longer supported, so I just dropped support for it here. We now support 12 and later, so I released a major version.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
-  script: yarn publish
+  provider: npm
+  email: open-source@goodeggs.com
+  api_token:
+    secure: 'mclmG0+zHQnQ3HlOCUEY3ioec8NT0jhOpk5fgN+vWnIlepFJ7cqD0LjhaXytnv4/WEQic25ym0mSD6OxFunJ+0h9CzMg3GCIIXAq7lrdQyxwFGn40kNSLlWyW+SniOE0zjb+8ne3zzMdXyquV0Lg43VflSn+5TzCRSsNjAQIXH6LySWqT/rVOzXiYvnV64A1XUCIWofCeDvrSbMmtXHqAQPXdd+N0zMN7y7jmQHjdoI4ZGcaec6Q+ygo/HfTST4anmyoe8EKfv9NFEBZx8rPlxoGJBl1HWoADl9OJdmUIaQon2bdn1oIyyhz1Jloxvp6uxZ/rdhkVaJ9qykPSbZd/du1W4itivPUcSRqIN95WxPgNXVieDLm02v5OD4F8pJV2FsvLbS+Zr8uBe5l+HgFdsFVcZ11XtimZD0lZtb9amfvfuaAn46GqpiygPANVDHt7+3wuw9rr8VlZ575MuPvz9N74fJ5hmcb/UGCiHBD4mQowLrpCvBAWPjDjli532XzXBu9Z3tHgeD6/leggDNiNlwrxL5sTUbF/UUngKD6k4I00oys6DJtwpPu02KYGMeY4sMPNQ7SlxoiRU/uAgld1iDUm5jAZTUAB8N59Hxp9Ib027v1B+xdSTOfcMW8XOxATkofEG8paAW1WBTgJBZLNSI6OxEPiZjxo9NQnAblnnU='
   skip_cleanup: true
   on:
     tags: true
-    branch: master
+    all_branches: true
```
Comment (whothey) [inline .travis.yml:29]: Looks like this will trigger deploys in any branch, right? 🤔 
If so, is this intended?

File: package.json [hunk 2]
```diff
@@ -21,12 +21,12 @@
   },
   "publishConfig": {
     "registry": "https://registry.npmjs.org/",
-    "always-auth": true
+    "access": "public"
   },
```
Comment (whothey) [inline package.json:25]: Can you consolidate this in the `.npmrc` file?

I didn't find anything on the [standards-and-best-practices](https://github.com/goodeggs/standards-and-best-practices) repo related to prefer dotfile/configuration files to package.json configs, so, the `.npmrc` should be fine? 😅

**Review by whothey — COMMENTED** [review]
LGTM!

Just a couple of questions.

File: .travis.yml (hunk 1 above)
Comment (serhalp) [reply .travis.yml:29]: Yes, that's how we set it up currently: https://goodeggs.atlassian.net/wiki/spaces/ENG/pages/459833453/Using+NPM+and+Yarn#How-to-configure-Travis-CI-to-automatically-publish-to-the-public-registry. (Don't worry, this won't trigger for a PR from a fork against a public repo.)

File: package.json (hunk 2 above)
Comment (serhalp) [reply package.json:25]: I'm following this guide to the letter: https://goodeggs.atlassian.net/wiki/spaces/ENG/pages/459833453/Using+NPM+and+Yarn#How-to-configure-a-public-module.

Keep in mind `.npmrc` applies to all uses of NPM, whereas `package.json#publishConfig` is config that overrides `.npmrc` (etc.) config specifically when in the context of publishing a package. `access` isn't meaningful in any other context, so we define it in the publish context.

Comment (whothey) [reply package.json:25]: Got it! Cool!

File: .travis.yml (hunk 1 above)
Comment (whothey) [reply .travis.yml:29]: Hahaha good to know!

**Review by whothey — APPROVED** [review]

**Pull request merged**